skribenten-backend/src/main/kotlin/no/nav/pensjon/brev/skribenten/services/BrevredigeringService.kt

```diff
@@ -145,4 +152,43 @@ class BrevredigeringService(
                 throw BrevbakerServiceException(response.error)
             }
         }
+
+    suspend fun ferdigstill(call: ApplicationCall, brevId: Long) {
+        val brevredigering = transaction { Brevredigering[brevId] }
+
+        val pesysData = hentPesysData(
+            call = call,
+            brevkode = brevredigering.brevkode,
+            saksId = brevredigering.saksId
+        )
+
+
+        brevbakerService.renderPdf(
+            call = call,
+            brevkode = brevredigering.brevkode,
+            brevdata = GeneriskRedigerbarBrevdata(
+                pesysData = pesysData.brevdata,
+                saksbehandlerValg = brevredigering.saksbehandlerValg,
+            ),
+            felles = pesysData.felles,
+            redigertBrev = brevredigering.redigertBrev.toMarkup()
+        ).onOk {
+            transaction {
+                Document.new {
+                    this.brevredigering = brevredigering
+                    pdf = ExposedBlob(it.file)
+                }
+            }
+        }.onError { error, statusCode ->
+            logger.error("En feil oppstod under rendering av PDF: $error", statusCode)
+            throw BrevbakerServiceException(error)
+        }
+    }
+
+    fun hentPdf(brevId: Long): ByteArray {
+        return transaction {
+            val brevredigering = Brevredigering.findById(brevId) ?: throw BrevbakerServiceException("Fant ikke brevredigering for brevId: $brevId")
+            brevredigering.document.first().pdf.bytes
```

Comment:
`first()` her vil kaste et unntak (som vil føre til 500 svar) om vi ikke har pdf for brevet det spørres om, og jeg tror at det vil være en del av vanlig flyt at vi sjekker om et brev har pdf eller ikke.

Så oppsummert sammen med forrige kommentar så tror jeg at det riktige i denne funksjonen er:

`Brevredigering.findById(brevId)?.document?.firstOrNull()?.pdf?.bytes`